### ISSUE TITLE
Enable highlighting of user specified lineages

### ISSUE THREAD

**Issue by andrewhill157** [CSE512-15S/a3-tdurham-ajh24-chiasson#4]
Tim already implemented the ability to highlight a lineage by name. We need to tie this to some sort of user input, such as hovering over or selecting a lineage in the tree and/or typing a name, etc. 

Any ideas here are welcome!

**Comment by tdurham86** [comment]
I added a text box and color picker for highlighting one lineage. We should definitely support highlighting multiple lineages, so perhaps we can talk about this when we meet tomorrow. If we implement this carefully, we can probably save ourselves some refactoring for the final project when we want to add gene expression-related highlighting.

**Comment by andrewhill157** [comment]
I added some styling to make it look a little better (with CSS), but didn't do too much here as it will probably change. The default value is red, but doesn't show on the face of the indicator for whatever reason. 

Didn't mess around with it too much as it may change.

**Comment by andrewhill157** [comment]
The current implementation is looking great -- just want to add that we should probably have the plot update when adding lineage highlighting so that if someone has it paused, the coloring appears immediately rather than when they switch to another frame. I would add this myself, but I feel like Tim might be editing this code and don't want to cause merge issues later.

**Comment by tdurham86** [comment]
Good idea -- I added this functionality to the lineage highlighting widgets.

**Comment by andrewhill157** [comment]
I think our current implementation is looking nice! One last thing that would be cool (if it is easy) as far as interactivity goes for assignment 3 would be the ability to click a node on the lineage tree and add a highlight (and potentially hover to highlight a cell). I'm not sure how easy/hard this former is with the custom search widget -- with a textbox, this would be a pretty easy addition using jquery I think? Any thoughts?

**Comment by andrewhill157** [comment]
I can take a crack at this if nobody else already is btw

**Comment by tdurham86** [comment]
I agree, I think that the mouseover highlight would be especially useful because it would allow you to locate specific cells from the tree in 3D space on the fly. I think we should try to find a mouseover highlighting strategy that works in conjunction with the dropdown-selected highlights.

As for clicking on a node to add a highlight, this would be ambitious because we would need to add a way for the user to specify which #lineage-picker\* the click should correspond to. Although perhaps it makes the most sense to do the highlight I mentioned above with a click instead of a mouseover -- it would reduce the processing load of spurious mouseover events.

**Comment by andrewhill157** [comment]
I think the clicking could be pretty simple if you add a new highlight or populate the last existing one if it unfilled. Anything more complicated than would be harder. 

If someone else wants to give the hover a try, that would be great. I probably don't have time to do both today.

**Comment by andrewhill157** [comment]
I didn't get far enough implementing this and I will be busy from now until 5. I essentially had started adding a callback to each node object:

```
var i = 0;
  var node = svg.append("g")
    .attr("class", "nodes")
    .selectAll(".node")
      .data(nodes, function(d) { return d.id || (d.id = ++i); })
        .enter().append("g")
        .attr("class", "node")
        .attr('onclick', 'highlightNodeLineage(event, this)')
        .append('circle')
          .attr('class', 'node-circle')
          .attr("r", 10)
          .attr("fill", "steelblue")
          .attr("transform", function(d) { 
            return "translate(" + 0 + "," + d.y + ")"; }) // 0 is required for x to make edges match up with nodes
          .call(position_node)
          .call(scale_radius)
```

The only added line above is: `.attr('onclick', 'highlightNodeLineage(event, this)')`

Here is the callback thus far:

```
function highlightNodeLineage(e, obj) {
    var cellName = $(obj).find('.text').text()

    var highlights = document.getElementsByClassName('selhi')
    lastHighlight = highlights.length

    console.log(highlights[lastHighlight -1 ])

    console.log(lastHighlight)
    if (highlights[lastHighlight - 1].value != '') {
        cloneLPDiv();
        highlights = $('selhi')
        highlights.find('.chosen-single').find('span').val(cellName).trigger("liszt:updated")
    } else {
        highlights[lastHighlight -1 ].val(cellName).trigger("liszt:updated")
    }
}
```

This is not working yet and has some debugging code. The main thing I could not figure out was how to actually set the value of the last highlight selector and trigger the update. I assumed you were using chosen, but the link I found (https://github.com/harvesthq/chosen/issues/605) that suggests using `.trigger("liszt:updated")` might be describing an older version.
